prow/config/cmd/generate.go

```diff
@@ -109,6 +128,52 @@ func main() {
 		}); err != nil {
 			exit(err, "walking through the meta config files failed")
 		}
+
+		// istio-private
```

Comment:
AFAICT the tool in `prow/config/cmd` is supposed to be a generic tool for generating general Prow jobs, and the tool in `prow/genjobs/` is used for transforming the generated Prow jobs to use for private repos. So can we have this change in `prow/genjobs/` instead?

Reply:
Thanks @chizhg. 

- `genjobs` is primarily for combining public jobs with private jobs (transforms) in order to produce cluster configs for istio-private jobs. 
- `prow/config/cmd/generate.go write` generates the public cluster config based on public cluster jobs. 
- `prow/config/cmd/generate.go branch` generates jobs for a new Istio branch, currently limited to only public jobs. 

This PR adds the ability to create private jobs for new Istio branches. After these are created, `make gen` can be run, which executes `genjobs` and `prow/config/cmd/generate.go write` in order to produce the cluster configs for both public and private jobs.

Reply:
Thanks @brian-avery .

It's still unclear to me why you need to update this tool and also import `genjobs` as a library... Is it going to generate the config files in `prow/config/istio-private_jobs`? If yes I feel it should be another tool to do this.

Reply:
Yeah, this is to generate the config files in `prow/config/istio-private_jobs`. I'm fine creating separate tools to generate public and private config. Regardless, the tool will have to pull in the Configuration struct as a library as `genjobs` should probably stay for translating the cluster config. It just means we will have to call two tools in release builder when we create the config.